product_tax_multicompany_default/models/product.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 2017 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# Copyright 2018 Vicent Cubells - Tecnativa <vicent.cubells@tecnativa.com>
+# Copyright 2023 Eduardo de Miguel - Moduon <edu@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    divergent_company_taxes = fields.Boolean(
+        string="Has divergent cross-company taxes",
+        compute="_compute_divergent_company_taxes",
+        compute_sudo=True,
+        help=(
+            "Does this product have divergent cross-company taxes? "
+            "(Only for multi-company products)"
+        ),
+    )
+
+    @api.depends("company_id", "taxes_id", "supplier_taxes_id")
+    def _compute_divergent_company_taxes(self):
+        """Know if this product has divergent taxes across companies."""
+        # Skip single-company products
+        if self.company_id or len(self.env["res.company"].search([]).ids) <= 1:
+            self.divergent_company_taxes = False
+            return
+        self.divergent_company_taxes = False
+        for one in self:
+            # A unique constraint in account.tax makes it impossible to have
+            # duplicated tax names by company
+
+            one.divergent_company_taxes = one._is_divergent_company_taxes(
+                "taxes"
+            ) or self._is_divergent_company_taxes("purchase")
+            if one.divergent_company_taxes:
+                continue
+
+    def _is_divergent_company_taxes(self, tax_type):
+        """Returns true or false if there are differences in product taxes.
+
+        :param one: product to be checked
+        :param tax_type: can be 'taxes' or 'purchase'
+        """
+        self.ensure_one()
+        all_companies = self.env["res.company"].search(
+            [
+                # Useful for tests, to avoid pollution
+                ("id", "not in", self.env.context.get("ignored_company_ids", [])),
+            ]
+        )
+        current_company = self.env.company
+
+        dict_product_taxes_bd = dict(self._origin._get_product_taxes(tax_type))
```

Comment:
"dict" word at the end. First the object, then the type.
Or make it more readable like the suggestion.

```suggestion
        company_tax_bd_map = dict(self._origin._get_product_taxes(tax_type))
```